consensus/src/round_manager.rs

```diff
@@ -1360,40 +1380,70 @@ impl RoundManager {
             return Ok(());
         }
 
-        let start = Instant::now();
-        order_vote_msg
-            .quorum_cert()
-            .verify(&self.epoch_state().verifier)
-            .context("[OrderVoteMsg QuorumCert verification failed")?;
-        counters::VERIFY_MSG
-            .with_label_values(&["order_vote_qc"])
-            .observe(start.elapsed().as_secs_f64());
+        // If a verified QC already exists in the cache, use it.
+        // Otherwise verify the QC from the order vote msg and then use it.
+        let verified_qc = match self
+            .order_vote_qc_cache
+            .get(&order_vote_msg.order_vote().ledger_info().hash())
```

Comment:
I see we're trying to avoid the second verification here, it seems easier to just cache the qc together inside the pending_order_vote and return it back than a separate cache?

Reply:
Done.